### ISSUE TITLE
Give a hint if kimageformats is not installed and a known extension fails to load

### ISSUE THREAD

**Issue by vistad** [nomacs/nomacs#1201]
Hello, I installed nomacs 3.17 because sometimes I need to see a few pictures simultaneously - and found that it does not support aviff.
But then I read the v3.16 supports aviff. 
I could not find neither the plugin nor explanation how to enable aviff support.
If you enabled it in v3.16 - why have you disabled it in v3.17???
Victor 
vstadnichenko@gmail.com

**Comment by novomesk** [comment]
AVIF format is supported (when corresponding Qt plug-in is installed too).

What is the **aviff** ? Try to attach the file which fails to open.

**Comment by vistad** [comment]
But is the avif plugin available?

**Comment by vistad** [comment]
I mean, is the Linux plugin available?

**Comment by leejuyuu** [comment]
@vistad Which distro are you using? The package to install also depends on the Qt version of your nomacs binary. You can find it in `? > About nomacs`.
On Arch I am using `qt6-imageformats`, and I believe it is `qt6-image-formats-plugins` for Qt6 on Ubuntu.

**Comment by novomesk** [comment]
The plugin is in kimageformats package on Linux.

On Windows, we use qt-avif-image-plugin

**Comment by vistad** [comment]
I am using Linux Mint. Could not find the avif plugin.

**Comment by leejuyuu** [comment]
Try `apt install kimageformat-plugins` if your nomacs uses Qt5, or `apt install kimageformat6-plugins` for Qt6.

**Comment by vistad** [comment]
Thanks for the tip! I will try it tomorrow. Why don't you write this
precious information on your website, right after the words that you have
enabled the avif support?
Best!

**Comment by scrubbbbs** [comment]
> Why don't you write this precious information on your website

Thanks, kimageformats-plugins will be added to the readme on github, on Linux/Unix systems in #1213. We are calling a lot of attention to it, so let's hope that package maintainers will add it to the "recommends" list of the package in a future update.

**Comment by vistad** [comment]
Thank you, I've already installed nomacs and kimageformats-plugins.
Everything works fine.
The bad thing is that the name "kimageformats-plugins" does not say that it
is related to avif or nomacs, so I should write down what it is and why I
need it - otherwise I'll not remember that it is related to avif and momacs.
Best of luck!

**Comment by scrubbbbs** [comment]
Should we have a dialog for certain file extensions when they aren't able to load? Something like,

"Sorry, I could not  load file.avif since AVIF is not supported. You can install `kimageformats-plugins` for Qt 6 to enable AVIF support. The package name may vary based on your distribution."

This might involve getting the `QImageReader::UnsupportedFormatError` result code from QImageReader so we could prevent showing it for corrupted files. We might also detect if the shared library is  present or not (kimg_avif.so).

**Comment by vistad** [comment]
Yes, that would be better than to leave a user nowhere. Tell them the
plugin name and where to get it.
